app/controllers/pages_controller.rb

```diff
@@ -1,9 +1,17 @@
 # frozen_string_literal: true
 
 class PagesController < SecureController
+  def create
+    p = Page.create!(params.require(:data).require(:attributes)
+                       .permit([:filename, :structure, :resource_id, :position]))
```

Comment:
You don't have to wrap the params to permit( ) in an array; none of the examples do.

You should consider configuring rubocop; it would complain about indentation here and it might alert about the array as well.